tests/unit/Commands/ContinuousIntegration/TravisCi/Config/ConfigWriterTest.php

```diff
@@ -0,0 +1,131 @@
+<?php
+
+namespace QuickStrapUnit\Commands\ContinuousIntegration\TravisCi\Config;
+
+use org\bovigo\vfs\vfsStream;
+use QuickStrap\Commands\ContinuousIntegration\TravisCi\Config\Config;
+use QuickStrap\Commands\ContinuousIntegration\TravisCi\Config\ConfigWriter;
+use Symfony\Component\Yaml\Yaml;
+
+class ConfigWriterTest extends \PHPUnit_Framework_TestCase
+{
+    /** @var  string */
+    private $configFile;
+    
+    /** @var ConfigWriter */
+    private $sut;
+
+    protected function setUp()
+    {
+        parent::setUp();
+
+        vfsStream::setup('application_path')
+            ->addChild($configFile = vfsStream::newFile('.travis.yml'));
+        
+        $this->configFile = $configFile->url();
+
+        $this->sut = new ConfigWriter();
+    }
+    
+    public function test_can_write_language_to_yml()
+    {
+        $config = new Config('php');
+
+        $actual = $this->sut->toYml($config);
+
+        self::assertEquals("language: php\n", $actual);
+    }
+    
+    public function test_can_write_php_versions_to_yml()
+    {
+        $config = new Config('php');
+        $config->setPhp(['7.0', '5.6']);
+
+        $actual = $this->sut->toYml($config);
+
+        self::assertEquals(<<<YML
+language: php
+php:
+    - '7.0'
+    - '5.6'
+
+YML
+, $actual);
+    }
+    
+    public function test_can_write_before_script_to_yml()
+    {
+        $config = new Config('php');
+        $config->setBeforeScript(['composer self-update', 'composer install']);
+
+        $actual = $this->sut->toYml($config);
+
+        self::assertEquals(<<<YML
+language: php
+before_script:
+    - 'composer self-update'
+    - 'composer install'
+
+YML
+, $actual);
+    }
+    
+    public function test_can_write_script_to_yml()
+    {
+        $config = new Config('php');
+        $config->setScript(['vendor/bin/phpunit', 'vendor/bin/behat']);
+
+        $actual = $this->sut->toYml($config);
+
+        self::assertEquals(<<<YML
+language: php
+script:
+    - vendor/bin/phpunit
+    - vendor/bin/behat
+
+YML
+, $actual);
+    }
+    
+    public function test_can_allow_failures_to_yml()
```

Comment:
the job of this class/method is to (given a hydrated Config object) generate a yml string (and write to a file, depending on which method you use). i care very much what the output of the yml generator is. without it, i can't be sure that the implementation detail (Yaml::dump() or Dumper::dump()) are doing the correct thing. nor am i free to write or provide another implementation that solves the same black box test in another fashion. would this test be considered a collaboration test, while a mocked dumper would be a contract test?

i'm not testing that Yaml does it's job, i'm testing that i'm using Yaml correctly to accomplish my goal. if the test is resilient enough, i could change the implementation of `toYml()` without breaking the test. 

without such a test, i am forced to guess or manually test that the system as a whole is functioning correctly. seems like a useful thing to keep around.